MembraneRTC/src/main/java/org/membraneframework/rtc/InternalMembraneRTC.kt

```diff
@@ -39,12 +32,15 @@ constructor(
     private val listener: MembraneRTCListener,
     @Assisted
     private val defaultDispatcher: CoroutineDispatcher,
-    audioDeviceModule: AudioDeviceModule,
     private val eglBase: EglBase,
     private val context: Context,
-    appContext: Context
-) : EventTransportListener, PeerConnection.Observer {
-    private var transport: EventTransport = connectOptions.transport
+    rtcEngineCommunicationFactory: RTCEngineCommunication.RTCEngineCommunicationFactory,
+    peerConnectionManagerFactory: PeerConnectionManager.PeerConnectionManagerFactory,
+    peerConnectionFactoryWrapperFactory: PeerConnectionFactoryWrapper.PeerConnectionFactoryWrapperFactory
```

Comment:
As you wrote in the PR description, this sounds suspicious. I don't know how to solve the problem but yeah, FactoryWrapperFactory is weird :thinking:

Reply:
I think it's just a weird name though. `PeerConnectionFactory` is a third-party dependency, we have no control over it, and we create wrapper around it - totally normal thing to do. And then we create a factory just for DI purpose (if not for one assisted dependency Dagger would probably create something like that automatically behind the scenes).

This wrapper saves us a lot of pain dealing with native android stuff when testing (I tried testing without it and it was a pain).